src/hooks/useTodos.ts

```diff
@@ -0,0 +1,101 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { Todo } from '../types/Todo';
+import * as todosService from '../api/todos';
+import { getCompletedTodos } from '../utils/getCompletedTodos';
+import { FilterType } from '../types/FilterType';
+import { filterTodos } from '../utils/filterTodos';
+import debounce from 'lodash.debounce';
+
+export const useTodos = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState<FilterType>('All');
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const setErrorDebounced = useCallback(debounce(setErrorMessage, 3000), []);
```

Comment:
The linter is alerting you that the function passed to useCallback has dependencies that are unknown.

Since debounce returns a new function each time, you should create the debounced function inside a useEffect or directly use a ref to store it.